### PR TITLE
Feature/filtrere ut etterbetaling saker

### PR DESCRIPTION
**Pull request by kariesta**
### 💰 Hva forsøker du å løse i denne PR'en
filterer ut saker som krever etterbetaling

### ✅ Checklist
_Har du husket alle punktene i listen?_
- [x ] Jeg har testet mine endringer i henhold til akseptansekriteriene 🕵️
- [ ] Jeg har skrevet tester. Hvis du ikke har skrevet tester, beskriv hvorfor under 👇

### 🤷‍♀ ️Hvor er det lurt å starte?
alt i ett :)

### 💬 Ønsker du en muntlig gjennomgang?
- [ ] Ja
- [x] Nei

### CODE REVIEW

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 1]
```diff
@@ -12,6 +14,7 @@ enum class FiltreringsreglerResultat(val beskrivelse: String) {
     MINDRE_ENN_5_MND_SIDEN_FORRIGE_BARN("Fødselshendelse: Mor har barn med mindre enn fem måneders mellomrom"),
     MOR_ER_IKKE_OVER_18("Fødselshendelse: Mor under 18 år"),
     MOR_HAR_VERGE("Fødselshendelse: Mor er umyndig"),
+    KREVER_ETTERBETALING("Fødselshendelse: saken krever etterbetaling"),
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:17]: Stor s i saken

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 2]
```diff
@@ -59,6 +64,16 @@ internal fun erBostNummer(personIdent: String): Boolean {
     return personIdent.substring(2, 3).toInt() > 1
 }
 
-internal fun barnetsFødselsdatoInnebærerIkkeEtterbetaling(): Boolean {
-    TODO("Sakene som krever etterbetaling skal filtreres ut. Det skal sikkert lages manuell sak da.")
+internal fun erBarnasFødselsdatoFør21IforrigeMåned(barnasFødselsdatoer: List<LocalDate>): Boolean {
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:67]: Om det er etter 21 i inneværende måned sjekker den vel om der er før 21 i inneværende måned?

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 3]
```diff
@@ -59,6 +64,16 @@ internal fun erBostNummer(personIdent: String): Boolean {
     return personIdent.substring(2, 3).toInt() > 1
 }
 
-internal fun barnetsFødselsdatoInnebærerIkkeEtterbetaling(): Boolean {
-    TODO("Sakene som krever etterbetaling skal filtreres ut. Det skal sikkert lages manuell sak da.")
+internal fun erBarnasFødselsdatoFør21InåværendeEllerforrigeMåned(barnasFødselsdatoer: List<LocalDate>): Boolean {
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:67]: Jeg tror egentlig innebærerBarnasFødselsdatoEtterbetaling gjør denne funksjonen mer forstålig

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 4]
```diff
@@ -59,6 +64,16 @@ internal fun erBostNummer(personIdent: String): Boolean {
     return personIdent.substring(2, 3).toInt() > 1
 }
 
-internal fun barnetsFødselsdatoInnebærerIkkeEtterbetaling(): Boolean {
-    TODO("Sakene som krever etterbetaling skal filtreres ut. Det skal sikkert lages manuell sak da.")
+internal fun innebærerBarnasFødselsdatoEtterbetaling(barnasFødselsdatoer: List<LocalDate>): Boolean {
+    val dagensDato = LocalDate.now()
+    val dagIMånedenForDagensDato = dagensDato.dayOfMonth
+    return when {
+        dagIMånedenForDagensDato < 21 && !barnasFødselsdatoer.all {
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:71]: Denne ble litt vanskelig å. Forstå kunne du ha endret det til for eksempel
`val alleBarnErFødtIInneværendeEllerForrigeMånde = !barnasFødselsdatoer.all {
            it.erFraInneværendeEllerForrigeMåned(dagensDato)
        }`
Og så brukt `dagIMånedenForDagensDato < 21 && !alleBarnErFødtIInneværendeEllerForrigeMånde`

Comment (halvorbmundal) [reply src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:71]: Eventuelt `val ingenBarnFødtFørForrigeMåned = barnasFødselsdatoer.all { it.erFraInneværendeEllerForrigeMåned(dagensDato) }`

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 5]
```diff
@@ -59,6 +64,16 @@ internal fun erBostNummer(personIdent: String): Boolean {
...
+            it.erFraInneværendeEllerForrigeMåned(dagensDato)
+        } -> true
+        dagIMånedenForDagensDato >= 21 && !barnasFødselsdatoer.all {
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:74]: Samme som over

File: src/test/kotlin/no/nav/familie/ba/sak/kjerne/automatiskVurdering/FiltreringsreglerUtilTest.kt [hunk 6]
```diff
@@ -195,6 +196,25 @@ class FiltreringsreglerUtilTest {
...
+        val filtreringsResultat =
+                evaluerFiltreringsregler(
+                        mor = søkerPerson,
+                        barnaFraHendelse = listOf(barn1Person),
+                        restenAvBarna = listOf(),
+                        morLever = true,
+                        barnaLever = true,
+                        morHarVerge = false
+                )
+        Assertions.assertEquals(FiltreringsreglerResultat.KREVER_ETTERBETALING, filtreringsResultat)
```
Comment (halvorbmundal) [inline src/test/kotlin/no/nav/familie/ba/sak/kjerne/automatiskVurdering/FiltreringsreglerUtilTest.kt:218]: Er det mulig å teste på edge casene? 
For eksempel før og etter inneværende måned for > 21 og før og etter Inneværende måned for >= 21

Comment (kariesta) [reply src/test/kotlin/no/nav/familie/ba/sak/kjerne/automatiskVurdering/FiltreringsreglerUtilTest.kt:220]: Tror det er godt dekket nå, lot tvillingene bli med også

Comment (kariesta) [reply src/test/kotlin/no/nav/familie/ba/sak/kjerne/automatiskVurdering/FiltreringsreglerUtilTest.kt:220]: Det kommer warnings fra tester pga. localDate-mocken men tror det skal gå fint, siden det ikke krasjer

File: src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt [hunk 7]
```diff
@@ -59,6 +64,19 @@ internal fun erBostNummer(personIdent: String): Boolean {
...
 
-internal fun barnetsFødselsdatoInnebærerIkkeEtterbetaling(): Boolean {
-    TODO("Sakene som krever etterbetaling skal filtreres ut. Det skal sikkert lages manuell sak da.")
+internal fun innebærerBarnasFødselsdatoEtterbetaling(barnasFødselsdatoer: List<LocalDate>): Boolean {
+    val dagensDato = LocalDate.now()
+    val dagensDatoErFør21Imåneden = dagensDato.dayOfMonth < 21
+    val barnErFødtFørForrigeMåned = barnasFødselsdatoer.any {
+        !it.erFraInneværendeEllerForrigeMåned(dagensDato)
+    }
+    val barnErFødtFørDenneMåned = barnasFødselsdatoer.any {
```
Comment (halvorbmundal) [inline src/main/kotlin/no/nav/familie/ba/sak/kjerne/automatiskvurdering/FiltreringsreglerUtil.kt:73]: barnErFødtFørDenneMåneden

**Review by halvorbmundal — APPROVED** [review]
